### PR TITLE
140189957 140189751 kh admin log in out

### PR DESCRIPTION
**Pull request by kheppenstall**
Functionality for an admin to log in and out.

@ski-climb Not sure how we want to handle the navbar and admin logout down the road.
*Right now the admin logging out is the same path as the user since it all it does is reset the session. 
*The navbar is based on `current_person` since there is no additional functionality an admin needs in the navbar.

@kheppenstall I don't see any downside to using the same path for the admin logout as the user logout.  Maybe we're missing something, but I agree with how you have set it up.  
As for the `current_person`, what comes to mind first is distinguishing the abilities/powers of a user vs. an admin.  For example, both users and admins can create issues, but right now we're setting the current_user for the issue, but not the admin.  And I think that's a good place to be right now.  I just added a story for Admin can create issue.  But we'll have to keep that kind of thing in mind.

Solid pull request.  Thanks for the solid work.

@ski-climb Thanks. Yea that makes sense. It could be as easy as changing `current_user` in the issues controller to `current_person` - since the relationships should be the same between them. If this totally blows up we could separate the navbars though and give different permissions more specifically.

### CODE REVIEW

**Comment by kheppenstall** [comment]
@ski-climb Ready for review.

**Comment by kheppenstall** [comment]
Styling: 
screen shot 2017-02-21 at 9 26 58 pm

File: app/services/permission.rb [hunk 1]
```diff
@@ -1,12 +1,14 @@
 class Permission
-  def initialize(user, controller, action)
-    @user       = user
+  def initialize(person, controller, action)
+    @person     = person
     @controller = controller
     @action     = action
   end
 
   def authorized?
```
Comment (ski-climb) [inline app/services/permission.rb]: Nice!  I like how this refactor came out.

File: spec/features/admin/login_spec.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      visit admin_login_path
+
+      fill_in 'email',    with: admin.email
+      fill_in 'password', with: 'password'
+      click_on 'Enter'
+
+      expect(current_path).to eq root_path
+      expect(page).to have_content "Welcome, #{admin.first_name}!"
+
+      session = page.get_rack_session
```
Comment (ski-climb) [inline spec/features/admin/login_spec.rb]: @kheppenstall Succinct and clean test.

File: spec/features/admin/login_spec.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      expect(current_path).to eq admin_login_path
+      expect(page).to have_content 'Invalid email or password.'
+
+      session = page.get_rack_session
+
+      expect(session).to_not have_key 'admin_id'
+    end
+  end
+
+  context 'as a logged in user' do
```
Comment (ski-climb) [inline spec/features/admin/login_spec.rb]: @kheppenstall Good edge case!

**Pull request merged**